exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1828 @@
+	case asset.Futures:
+		orderID, err = by.ReplaceActiveFuturesOrders(ctx, action.Pair, action.OrderID, action.ClientOrderID, "", "", action.Amount, action.Price, 0, 0)
+	case asset.USDCMarginedFutures:
+		// TODO: take suggestion related to orderFilter. option accepted by bybit Order/StopOrder
+		orderID, err = by.ModifyUSDCOrder(ctx, action.Pair, "Order", action.OrderID, action.ClientOrderID, action.Price, action.Amount, 0, 0, 0, 0, 0)
+	default:
+		err = fmt.Errorf("%s %w", action.AssetType, asset.ErrNotSupported)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := action.DeriveModifyResponse()
+	if err != nil {
+		return nil, err
+	}
+	resp.OrderID = orderID
+	return resp, nil
+}
+
+// CancelOrder cancels an order by its corresponding ID number
+func (by *Bybit) CancelOrder(ctx context.Context, ord *order.Cancel) error {
+	if err := ord.Validate(ord.StandardCancel()); err != nil {
+		return err
+	}
+
+	var err error
+	switch ord.AssetType {
+	case asset.Spot:
+		_, err = by.CancelExistingOrder(ctx, ord.OrderID, ord.ClientOrderID)
+	case asset.CoinMarginedFutures:
+		_, err = by.CancelActiveCoinFuturesOrders(ctx, ord.Pair, ord.OrderID, ord.ClientOrderID)
+	case asset.USDTMarginedFutures:
+		_, err = by.CancelActiveUSDTFuturesOrders(ctx, ord.Pair, ord.OrderID, ord.ClientOrderID)
+	case asset.Futures:
+		_, err = by.CancelActiveFuturesOrders(ctx, ord.Pair, ord.OrderID, ord.ClientOrderID)
+	case asset.USDCMarginedFutures:
+		_, err = by.CancelUSDCOrder(ctx, ord.Pair, "Order", ord.OrderID, ord.ClientOrderID)
+	default:
+		return fmt.Errorf("%s %w", ord.AssetType, asset.ErrNotSupported)
+	}
+	return err
+}
+
+// CancelBatchOrders cancels orders by their corresponding ID numbers
+func (by *Bybit) CancelBatchOrders(ctx context.Context, orders []order.Cancel) (order.CancelBatchResponse, error) {
+	return order.CancelBatchResponse{}, common.ErrNotYetImplemented
+}
+
+// CancelAllOrders cancels all orders associated with a currency pair
+func (by *Bybit) CancelAllOrders(ctx context.Context, orderCancellation *order.Cancel) (order.CancelAllResponse, error) {
+	if err := orderCancellation.Validate(); err != nil {
+		return order.CancelAllResponse{}, err
+	}
+
+	status := "success"
+	var cancelAllOrdersResponse order.CancelAllResponse
+	cancelAllOrdersResponse.Status = make(map[string]string)
+	switch orderCancellation.AssetType {
+	case asset.Spot:
+		activeOrder, err := by.ListOpenOrders(ctx, orderCancellation.Pair.String(), "", 0)
+		if err != nil {
+			return cancelAllOrdersResponse, err
+		}
+
+		var orderType, side string
+		if orderCancellation.Type != order.UnknownType {
+			orderType = orderCancellation.Type.String()
+		}
+		if orderCancellation.Side != order.UnknownSide {
+			side = orderCancellation.Side.Title()
+		}
+
+		successful, err := by.BatchCancelOrder(ctx, orderCancellation.Pair.String(), side, orderType)
+		if !successful {
+			return cancelAllOrdersResponse, fmt.Errorf("failed to cancelAllOrder")
+		}
```

Comment:
I'm thinking it would be best to do the following:

```go
successful, err := by.BatchCancelOrder(ctx, orderCancellation.Pair.String(), side, orderType)
		if !successful {
			status = "failed"
		}
		if err != nil {
			status = err.Error()
		}
		for i := range activeOrder {
			cancelAllOrdersResponse.Status[strconv.FormatInt(activeOrder[i].OrderID, 10)] = status
		}
```
That way we have a generic "this failed to cancel" message, but if more detail is provided in the error, it will use that instead and we still get the itemised response to attempting to cancel all orders

Reply:
Okay. Sounds good. I will make these changes now.